src/LegendRenderer/SvgOutput.ts

```diff
@@ -85,32 +96,59 @@ export default class SvgOutput extends AbstractOutput {
     drawRect: boolean
   ) {
     if (drawRect) {
-      this.currentContainer?.append('rect')
+      this.currentContainer
+        ?.append('rect')
         .attr('x', x)
         .attr('y', y)
         .attr('width', imgWidth)
         .attr('height', imgHeight)
         .style('fill-opacity', 0)
         .style('stroke', 'black');
     }
-    this.currentContainer?.append('svg:image')
+    this.currentContainer
+      ?.append('svg:image')
       .attr('x', x)
       .attr('y', y)
       .attr('width', imgWidth)
       .attr('height', imgHeight)
       .attr('href', dataUrl);
     this.root?.attr('xmlns', 'http://www.w3.org/2000/svg');
     return Promise.resolve();
-  };
+  }
 
   generate(finalHeight: number) {
     const nodes = this.root?.selectAll('g.legend-item');
-    this.shortenLabels(nodes, this.maxColumnWidth || 0, this.legendItemTextSize);
+    let finalWidth = this.size[0];
+    if (typeof this.maxColumnWidth === 'number') {
```

Comment:
Before, `this.shortenLabels` was called regardless of the type of `this.maxColumnWidth`. Now, labels will only be shortened if `this.maxColumnWidth` is numeric (i.e. no shortening when `maxColumnWidth` is undefined). Is this intentional behaviour?

If not, I would propose to change this section to, so that the old behaviour is restored:

```
if (this.maxColumnwidth === 'fit-content') {
  // ...
} else {
  // ...
}
```